src/Compilers/CSharp/Portable/BoundTree/DecisionTree.cs

```diff
@@ -53,7 +53,7 @@ public static DecisionTree Create(BoundExpression expression, TypeSymbol type, S
                 expression = new BoundLocal(expression.Syntax, temp, null, type);
             }
 
-            if (expression.Type.CanBeAssignedNull())
```

Comment:
It looks like this helper is also used in localrewriter_patternswitchstatement.cs, is it possible that that code should be adjusted as well?